packages/studio/src/components/PillPicker/PillPicker.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useRef, useState } from "react";
+import { ReactComponent as EmbedIcon } from "../../icons/embed.svg";
+import { useRootClose } from "@restart/ui";
+import classNames from "classnames";
+
+interface PillPickerProps {
+  availableItems: string[];
+  selectItem: (item: string) => void;
+  disabled?: boolean;
+  getDisplayName?: (item: string) => string;
+}
+
+/**
+ * An icon that displays a dropdown of available items when clicked.
+ */
+export default function PillPicker({
+  availableItems,
+  selectItem,
+  disabled,
+  getDisplayName,
+}: PillPickerProps) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useRootClose(containerRef, () => {
+    setIsOpen(false);
+  });
+
+  const toggleOpen = useCallback(() => {
+    if (disabled) {
+      return;
+    }
+    setIsOpen((isOpen) => !isOpen);
+  }, [disabled]);
+
+  const handleSelect = useCallback(
+    (item: string) => {
+      toggleOpen();
```

Comment:
I think we may want to keep the dropdown open to allow them to select multiple things in one go. This situation is a bit different than the Insert button. In that case, we assumed adding multiple Components in one go was uncommon. For Stream Scope selection, I think adding multiple entity ids or entity types will be common.

Reply:
Updated! This also affects the dropdown for the Tailwind class prop input, but I don't think that's an issue since I could see people wanting to add multiple Tailwind classes at once